atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -239,6 +243,11 @@ boolean allSafeForLogging() {
         return false;
     }
 
+    @Value.Default
+    boolean useDialogue() {
```

Comment:
This class is directly changed in code, meaning that rollout might be harder?

When do rollouts like these, we make changes to AtlasDbRuntimeConfig (which clients wire into their service's config), and specifically I think something like RemotingClientConfig (which was used to rollout http-remoting -> CJR switch) might be useful to you here. I am happy to start with a DialogueClientConfig that has a boolean flag to just enable it, if we use it for testing in a dev environment. Having a dedicated DialogueClientConfig should hopefully make it easier to remove later.

Reply:
Hmm not sure I understand what the issue with adding the flag to the transactionManager directly. It seems just as easy to remove/change the default and avoids an extra level of indirection

Reply:
Ok, I didn't explain properly.

What it means is that you need any application that picks this up to push special code that sets this flag.

They can either:
1. Risk that everything will be ok and just hard set it. If they hit issues, they need to rollback in production, then revert your change and cut a new release so they're not blocking develop.
2. Add custom configuration block to their config so that you can switch this via config.

I don't want anyone to do 1, or 2, because AtlasDbRuntimeConfiguration is how changeable parameters to TransactionManagers should be handled.